File: Python Programs/task_Scheduling.py
def is_feasible(tasks, workers, target):
    workers_used = 1
    current_workload = 0
    
    for task in tasks:
        if current_workload + task > target:
            workers_used += 1
            current_workload = task
            if workers_used > workers:
                return False
        else:
            current_workload += task
    
    return True

def min_time(tasks, workers):
    left = max(tasks)  # Minimum possible time is the maximum task duration
    right = sum(tasks)  # Maximum possible time is the total sum of task durations
    
    while left < right:
        mid = (left + right) // 2
        if is_feasible(tasks, workers, mid):
            right = mid
        else:
            left = mid + 1
    
    return left

# Input
N = int(input().strip())  # Number of tasks
tasks = list(map(int, input().strip().split()))  # Task durations
K = int(input().strip())  # Number of workers

# Calculate and output the minimum time required
print(min_time(tasks, K))

File: Java Programs/task_scheduling.java
import java.util.Scanner;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int[] tasks = new int[n];
        for (int i = 0; i < n; i++) {
            tasks[i] = input.nextInt();
        }
        int k = input.nextInt();
        
        int minTime = minTimeToComplete(tasks, k);
        System.out.println(minTime);
    }
    
    public static int minTimeToComplete(int[] tasks, int k) {
        int start = Arrays.stream(tasks).max().getAsInt();
        int end = Arrays.stream(tasks).sum();
        
        while (start < end) {
            int mid = start + (end - start) / 2;
            if (isPossible(tasks, mid, k)) {
                end = mid;
            } else {
                start = mid + 1;
            }
        }
        return start;
    }
    
    public static boolean isPossible(int[] tasks, int maxTime, int k) {
        int workers = 1;
        int totalTime = 0;
        
        for (int task : tasks) {
            totalTime += task;
            if (totalTime > maxTime) {
                totalTime = task;
                workers++;
            }
        }
        
        return workers <= k;
    }
}